src/helpers/soroban.ts

```diff
@@ -284,125 +192,127 @@ export const getTokenSymbol = async (
     .setTimeout(TimeoutInfinite)
     .build();
 
-  const result = await simulateTx<string>(tx, decoders.bytesN, server);
+  const result = await simulateTx<string>(tx, server);
   return result;
 };
 
 export const buildContractAuth = async (
-  auths: string[],
-  signerKeypair: Keypair,
+  authEntries: xdr.SorobanAuthorizationEntry[],
+  signerPubKey: string,
   networkPassphrase: string,
   contractID: string,
   server: Server,
+  kit: StellarWalletsKit,
 ) => {
-  const contractAuths = [];
+  const signedAuthEntries = [];
 
-  if (auths) {
-    for (const authStr of auths) {
-      const contractAuth = xdr.ContractAuth.fromXDR(authStr, "base64");
-
-      if (contractAuth.addressWithNonce()) {
-        const authAccount = contractAuth
-          .addressWithNonce()!
+  if (authEntries.length) {
+    for (const entry of authEntries) {
+      if (entry.credentials().switch().name === "sorobanCredentialsAddress") {
```

Comment:
it would be slightly better avoid the magic constant:

```suggestion
      if (entry.credentials().switch() == xdr.SorobanCredentialsType.sorobanCredentialsAddress()) {
```

Reply:
ah nice thanks, noting that you have to cast this right now because `string != xdr.SorobanCredentialsType`, those could probably be string enums but iirc those are generated xdr classes 😅

Reply:
ah nice, thanks!

Reply:
fixed in bb8144443b48b84d73d1bfcfe1bacccf20c91cde